src/parser.ts

```diff
@@ -3499,6 +3521,13 @@ export class Parser extends DiagnosticEmitter {
         tn.skip(Token.BAR);
         let type = this.parseType(tn);
         if (!type) return null;
+        if (this.checkRecurseDefinitionForType(name.text, type)) {
+          this.error(
+            DiagnosticCode.Not_implemented_0,
+            tn.range(), "recursion in type aliases"
+          );
```

Comment:
Would TS2456 fit here? `Type alias '{0}' circularly references itself.`

Reply:
`TS2456` only appear for such case `type T = T | null` but success for `type T = () => T` for example.

Reply:
There's also 2310: `"Type '{0}' recursively references itself as a base type."`. In general it seems that any of these two is preferable over a "not implemented", even if it doesn't match TS exactly.

Reply:
Technically this makes sense, yeah, was mostly thinking about user expectation. Maybe you are right, leaving it up to you :)